tesserocr/tesserocr.pyx

```diff
@@ -40,13 +40,26 @@ from cython.operator cimport preincrement as inc, dereference as deref
 from cpython.version cimport PY_MAJOR_VERSION
 
 
-cdef bytes _b(s):
-    if PY_MAJOR_VERSION > 3:
+cdef bytes _b(object s):
+    """Convert string to bytes for cross-Python compatibility.
+
+    Args:
+        s: Input string or bytes object
+
+    Returns:
+        bytes: UTF-8 encoded bytes object
+    """
+    if PY_MAJOR_VERSION >= 3:
         if isinstance(s, str):
             return s.encode('UTF-8')
-    elif isinstance(s, unicode):
-        return s.encode('UTF-8')
-    return s
+        elif isinstance(s, bytes):
+            return s
+    else:
+        # Python 2: unicode needs encoding
+        if isinstance(s, unicode):
+            return s.encode('UTF-8')
+        elif isinstance(s, str):
+            return s
```

Comment:
Since you switched to cdef here, do you think it would make sense to use overloading instead? (not sure if this even works)

```
cdef bytes _b(bytes s):
    return s
    
cdef bytes _b(unicode s):  # matches str in py3
    return s.encode("utf-8")
```

Should work for both py2 and 3 (although I already dropped support for py2 but it may still be working otherwise so we can keep unicode since cython understands it for both py2 and py3)